File: app/src/main/java/com/biryanistudio/todo/userinterface/ActionTextActivity.java
package com.biryanistudio.todo.userinterface;

import android.app.Activity;
import android.content.Intent;
import android.os.Build;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.annotation.RequiresApi;

import com.biryanistudio.todo.database.DbTransactions;

public class ActionTextActivity extends Activity {

    @RequiresApi(api = Build.VERSION_CODES.M)
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        String text = getIntent().getCharSequenceExtra(Intent.EXTRA_PROCESS_TEXT).toString().trim();
        DbTransactions.writeTask(this, text);
        UiUtils.createNotification(this, text);
        finish();
    }
}

File: app/src/main/java/com/biryanistudio/todo/userinterface/ActionTextActivity.kt
package com.biryanistudio.todo.userinterface

import android.app.Activity
import android.content.Intent
import android.os.Build
import android.os.Bundle
import android.support.annotation.RequiresApi
import com.biryanistudio.todo.TodoApplication
import com.biryanistudio.todo.database.TodoItem
import com.biryanistudio.todo.database.TransactionsHelper

class ActionTextActivity : Activity() {

    @RequiresApi(api = Build.VERSION_CODES.M)
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val text = intent.getCharSequenceExtra(Intent.EXTRA_PROCESS_TEXT).toString()
                .trim { it <= ' ' }
        TransactionsHelper.addItem(TodoItem(task = text, completed = 0,
                timestamp = System.currentTimeMillis()))
        TodoApplication.createNotification(this, text)
        finish()
    }
}
